### ISSUE TITLE
Use UNUserNotificationCenter class 

### ISSUE THREAD

**Issue by andrewtremblay-pear** [react-native-push-notification-ios/push-notification-ios#25]
# Feature Request

To be more future future-proof against iOS versions, we should be using the `UNUserNotificationCenter` instead of deprecated methods. 
https://developer.apple.com/documentation/usernotifications/unusernotificationcenter?language=objc

## Why it is needed

For example, `cancelAllLocalNotifications` is deprecated as of iOS 10.
https://developer.apple.com/documentation/uikit/uiapplication/1622990-cancelalllocalnotifications?language=objc

At some point, this will become a no-op.

## Possible implementation

This will not be a simple 1:1 replacement. In the case of cancelAllLocalNotifications, UNUserNotificationCenter now has two calls:

```objective-c
UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
[center removeAllDeliveredNotifications];
[center removeAllPendingNotificationRequests];

```

**Comment by holden-caulfield** [comment]
I could definitively use a migration like this on my project. Currently I need to detect when a user taps on a notification, and the current behavior of this library is:
- If the application was killed, then when it is launched by a notification tap I can get the notification that launched the app with `getInitialNotification`
- If the application is on the background, most of the times I get the `notification` event, which gets called when a notification is _received_ called *again* when it is tapped. 

So I'm now implementing partially UNUserNotificationCenter to handle this case, outside the library:
```
- (void)userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNotificationResponse:(UNNotificationResponse *)response withCompletionHandler:(void (^)(void))completionHandler {
  completionHandler();
}
```

As a weird side effect, the `notification` event does not get called a second time anymore since I've implemented this handler (though I guess that's how it is supposed to be as Apple says it _may_ but not necessarily _will_ call `didReceiveNotification` on tap)

So, it would be great if after migrating to `UNUserNotificationCenter` there is one more valid event to `PushNotificationiOS.addEventListener` called `notificationTap` or something like that

**Comment by nbolender** [comment]
Implemented in [v1.7.0](https://github.com/react-native-push-notification-ios/push-notification-ios/tree/v1.7.0)

**Comment by Naturalclar** [comment]
Closing as methods for UNNotificationCenter has been implemented in v1.7.0
